lib/application/utils/bottom_sheet.dart

```diff
@@ -0,0 +1,134 @@
+//
+// Relevant issues:
+// - https://github.com/flutter/flutter/issues/34111
+// - https://github.com/flutter/flutter/issues/45009
+import 'package:flutter/material.dart';
+import 'package:flutter_hooks/flutter_hooks.dart';
+import 'package:layoutr/common_layout.dart';
+import 'package:memo/application/constants/dimensions.dart' as dimens;
+import 'package:memo/application/theme/theme_controller.dart';
+
+extension BottomSheetExtension on BuildContext {
+  /// Wraps a [showModalBottomSheet] behavior that snaps its content based on [child] size
+  ///
+  /// If [isDismissible] is `false`, all drag interactions are disabled and the caller must handle its dismissal, like
+  /// some close button or any other action that will eventually dismiss this modal. Also, no dragIndicator is drawn,
+  /// to corroborate with the no-dragging behavior.
+  /// 
+  /// The [child] must have a size to fit whitin the SingleChildScrollView
+  Future<T?> showDraggableScrollableModalBottomSheet<T>({
+    required Widget child,
+    bool isDismissible = true,
+    Color? backgroundColor,
+    String? title,
+    Widget? leadingWidget,
+    Widget? trailingWidget,
+  }) {
+    final dragIndicator = Container(
+      width: dimens.dragIndicatorWidth,
+      height: dimens.dragIndicatorHeight,
+      decoration: const BoxDecoration(
+        color: Colors.grey,
+        borderRadius: BorderRadius.all(Radius.circular(dimens.bottomSheetDragIndicatorRadius)),
+      ),
+    ).withSymmetricalPadding(this, vertical: Spacing.small);
+
+    final hasHeaderItems = leadingWidget != null || title != null || trailingWidget != null;
+
+    final header = Column(
+      crossAxisAlignment: CrossAxisAlignment.stretch,
+      children: [
+        if (isDismissible) Center(child: dragIndicator),
+        if (hasHeaderItems)
+          Stack(
+            alignment: Alignment.center,
+            children: [
+              if (leadingWidget != null) Align(alignment: Alignment.centerLeft, child: leadingWidget),
+              if (title != null) Align(child: Text(title, style: Theme.of(this).textTheme.headline6)),
+              if (trailingWidget != null) Align(alignment: Alignment.centerRight, child: trailingWidget),
+            ],
+          ).withSymmetricalPadding(this, vertical: Spacing.small, horizontal: Spacing.medium)
+      ],
+    );
+
+    return showModalBottomSheet<T>(
+      context: this,
+      isDismissible: isDismissible,
+      enableDrag: isDismissible,
+      isScrollControlled: true,
+      builder: (context) {
+        return _ModalBottomSheet(
+          header: header,
+          isDismissible: isDismissible,
+          title: title,
+          leadingWidget: leadingWidget,
+          trailingWidget: trailingWidget,
+          backgroundColor: backgroundColor,
+          child: child,
+        );
+      },
+    );
+  }
+}
+
+class _ModalBottomSheet extends HookWidget {
```

Comment:
I did it this way because the useTheme() needs to be called directly within the build method of a HookWidget.

Is it clear? I couldn't figure out another solution, any suggestion? @ggirotto

Reply:
You can read the `themeController` from the widgets tree, since it's a provider and was previously injected by doing:

```dart
final theme = context.read(themeController.state);
```

In this case, since you're inside a `BuildContext` extension, you may just:

```dart
final theme = read(themeController.state);
```

Then you don't need this separate Widget anymore